x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/immutablestate/ILMImmutableStateHandlerProvider.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ilm.immutablestate;
+
+import org.elasticsearch.immutablestate.ImmutableClusterStateHandler;
+import org.elasticsearch.immutablestate.ImmutableClusterStateHandlerProvider;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * ILM Provider implementation for the {@link ImmutableClusterStateHandlerProvider} service interface
+ */
+public class ILMImmutableStateHandlerProvider implements ImmutableClusterStateHandlerProvider {
+    private static final Set<ImmutableClusterStateHandler<?>> handlers = ConcurrentHashMap.newKeySet();
+
+    @Override
+    public Collection<ImmutableClusterStateHandler<?>> handlers() {
+        return handlers;
+    }
+
+    public static void handler(ImmutableClusterStateHandler<?> handler) {
```

Comment:
I think does need to be invoked and should provide a `ImmutableLifecycleAction` instance, right?

Reply:
Ah nice catch, I should've brought in this change in this PR. https://github.com/elastic/elasticsearch/pull/86224/files#diff-245140b1f56e32ce2f87c455562395af21e0b6730984b7077dd236b7086e5164R275